core/src/Kokkos_BitManipulation.hpp

```diff
@@ -259,7 +259,7 @@ KOKKOS_IMPL_HOST_FUNCTION T byteswap_builtin_host(T x) noexcept {
   } else if constexpr (sizeof(T) == 8) {
     return __builtin_bswap64(x);
   } else if constexpr (sizeof(T) == 16) {
-#if __has_builtin(__builtin_bswap128)
+#if defined (__has_builtin) && __has_builtin(__builtin_bswap128)
```

Comment:
I don't think that would work, all of it would get evaluated

Reply:
Seems to work now.